include/ocpp/v201/smart_charging.hpp

```diff
@@ -48,13 +48,28 @@ namespace conversions {
 /// \brief Converts the given ProfileValidationResultEnum \p e to human readable string
 /// \returns a string representation of the ProfileValidationResultEnum
 std::string profile_validation_result_to_string(ProfileValidationResultEnum e);
+
+/// \brief Converts the given ProfileValidationResultEnum \p e to a OCPP reasonCode.
+/// \returns a reasonCode
+std::string profile_validation_result_to_reason_code(ProfileValidationResultEnum e);
 } // namespace conversions
 
 std::ostream& operator<<(std::ostream& os, const ProfileValidationResultEnum validation_result);
 
+class SmartChargingHandlerInterface {
+public:
+    virtual ~SmartChargingHandlerInterface();
+
+    virtual ProfileValidationResultEnum validate_profile(ChargingProfile& profile, int32_t evse_id) = 0;
+
+    virtual SetChargingProfileResponse add_profile(int32_t evse_id, ChargingProfile& profile) = 0;
```

Comment:
Would it be a hassle to flip one of these two parameters so they are consistent? There were previous PR with this comment.

Reply:
I thought you were handling that in a cleanup PR against Everest main?

Reply:
I have it locally but have not pushed it up yet. Since this is active and near ready I think it would be better to have it here then in a separate PR.

Reply:
Done